ibis/pandas/execution/selection.py

```diff
@@ -231,10 +255,11 @@ def _compute_predicates(table_op, predicates, data, scope, **kwargs):
                 new_data = data.loc[:, mapping.keys()].rename(columns=mapping)
             else:
                 new_data = data
-            additional_scope[root_table] = new_data
+            scope.merge_scope(
```

Comment:
We should try to not change the original code structure here.

This can be 
```
additional_scope = Scope()
for root_table in root_tables:
    additional_scope.merge_scope(...)

scope.merge_scope(additional_scope)
```

Reply:
It seems identical but I followed your advice and the original code structure.